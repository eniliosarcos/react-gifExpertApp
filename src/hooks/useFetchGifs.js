import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifts";

export const useFetchGifs = (collection) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(collection).then(gifs => {

            setTimeout(() => {
                
                setState({
                    data: gifs,
                    loading: false
                })
            }, 300);
        });
    }, [collection]);

    return state; // {data: [], loading: true};
}
