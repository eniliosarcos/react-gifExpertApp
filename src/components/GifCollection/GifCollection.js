import PropTypes from "prop-types";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import GifImg from "../GifImg/GifImg";
import "./GifCollection.css";

const GifCollection = ({ collection }) => {

  const {data:gifs, loading} = useFetchGifs(collection);

  return (
    <>
      <h3 className="animate__animated animate__fadeInDown animate__faster">{collection}</h3>
      { loading && <p className="animate__animated animate__flash animate__slow animate__infinite">Cargando...</p>}
      <div className="grid">
        {
            gifs.map(({ id, title, url }) => <GifImg key={id} title={title} url={url}></GifImg>)
        }
      </div>
    </>
  );
};

GifCollection.propTypes = {};

export default GifCollection;
