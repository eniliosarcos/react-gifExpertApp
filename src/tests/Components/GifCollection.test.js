import { shallow } from "enzyme";
import GifCollection from "../../components/GifCollection/GifCollection";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el componente GifCollection', () => {

    test('Debe de mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifCollection collection="Dragon ball"></GifCollection>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes con useFetchGifs', () =>{
        
        const gifs = [{
            id:'ABC',
            url: 'https//localhost/tortuga.jpg',
            title: 'tortuga'
        },{
            id:'ABC123',
            url: 'https//localhost/papelon.jpg',
            title: 'papelon'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifCollection collection="Dragon ball"></GifCollection>);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifImg').length).toBe(gifs.length);
    });
})
