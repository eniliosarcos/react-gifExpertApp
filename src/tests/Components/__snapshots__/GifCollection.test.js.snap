// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifCollection Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown animate__faster"
  >
    Dragon ball
  </h3>
  <p
    className="animate__animated animate__flash animate__slow animate__infinite"
  >
    Cargando...
  </p>
  <div
    className="grid"
  />
</Fragment>
`;
